bin/testEnvLocalK8S.sh

```diff
@@ -0,0 +1,116 @@
+LOG_DIR=${LOG_DIR:-${OUT}/log}
+
+EASYRSA_DIR=$OUT/easy-rsa-master/easyrsa3
+EASYRSA=$EASYRSA_DIR/easyrsa
+
+# Ensure k8s certificats - if not found, download easy-rsa and create k8s certs
+function ensureK8SCerts() {
+    if [ -f ${CERTDIR}/apiserver.key ] ; then
+        return
+    fi
+
+    mkdir -p ${CERTDIR}
+    pushd $OUT
+    curl -L -O https://storage.googleapis.com/kubernetes-release/easy-rsa/easy-rsa.tar.gz
+    tar xzf easy-rsa.tar.gz
+    cd easy-rsa-master/easyrsa3
+
+    ./easyrsa init-pki
+    ./easyrsa --batch "--req-cn=${MASTER_IP}@`date +%s`" build-ca nopass
+    ./easyrsa --subject-alt-name="IP:${MASTER_IP},""IP:${MASTER_CLUSTER_IP},""DNS:kubernetes,""DNS:kubernetes.default,""DNS:kubernetes.default.svc,""DNS:kubernetes.default.svc.cluster,""DNS:kubernetes.default.svc.cluster.local" --days=10000 build-server-full server nopass
+
+    cp pki/ca.crt ${CERTDIR}/k8sca.crt
+    cp pki/issued/server.crt ${CERTDIR}/apiserver.crt
+    cp pki/private/server.key ${CERTDIR}/apiserver.key
+    popd
+}
+
+# Get dependencies needed for tests. Only needed once.
+# The docker builder image should include them.
+function getDeps() {
+   mkdir -p $TOP/bin
+   if [ ! -f $TOP/bin/kubectl ] ; then
+     curl -Lo $TOP/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl && chmod +x $TOP/bin/kubectl
+   fi
+   if [ ! -f $TOP/bin/kube-apiserver ] ; then
+     curl -Lo ${TOP}/bin/kube-apiserver https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kube-apiserver && chmod +x ${TOP}/bin/kube-apiserver
+   fi
+   if [ ! -f $TOP/bin/etcd ] ; then
+     go get github.com/coreos/etcd/cmd/etcd
+   fi
+   if [ ! -f $TOP/bin/envoy ] ; then
+    # Original circleci - replaced with the version in the dockerfile, as we deprecate bazel
+    #ISTIO_PROXY_BUCKET=$(sed 's/ = /=/' <<< $( awk '/ISTIO_PROXY_BUCKET =/' WORKSPACE))
+    #PROXYVERSION=$(sed 's/[^"]*"\([^"]*\)".*/\1/' <<<  $ISTIO_PROXY_BUCKET)
+    PROXYVERSION=$(grep envoy-debug pilot/docker/Dockerfile.proxy_debug  |cut -d: -f2)
+    PROXY=debug-$PROXYVERSION
+    pushd $OUT
+    # TODO: Use circleci builds
+    curl -Lo - https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xz
+    cp usr/local/bin/envoy $TOP/bin/envoy
+    popd
+
+   fi
+}
+
+# No root required, run local etcd and kube apiserver for tests.
+function startLocalApiserver() {
+    ensureK8SCerts
+    getDeps
+
+    mkdir -p ${LOG_DIR}
+
+    ${TOP}/bin/etcd > ${LOG_DIR}/etcd.log &
+    echo $! > $LOG_DIR/etcd.pid
+
+    ${TOP}/bin/kube-apiserver --etcd-servers http://127.0.0.1:2379 \
+        --client-ca-file ${CERTDIR}/k8sca.crt \
+        --requestheader-client-ca-file ${CERTDIR}/k8sca.crt \
+        --tls-cert-file ${CERTDIR}/apiserver.crt \
+        --tls-private-key-file ${CERTDIR}/apiserver.key \
+        --service-cluster-ip-range 10.99.0.0/16 \
+        --port 8080 -v 2 --insecure-bind-address 0.0.0.0 \
```

Comment:
The port - yes, I'm planning to do this - current code is cut&pasted from circleci, but for running on 
local machine it probably needs more customization. However not top priority, after it works we can
fine tune it... (and better yet - this can be moved into the new test framework !)